APMrover2/failsafe.cpp

```diff
@@ -68,7 +68,7 @@ void Rover::failsafe_trigger(uint8_t failsafe_type, bool on)
         control_mode != &mode_rtl &&
         control_mode != &mode_hold) {
         failsafe.triggered = failsafe.bits;
-        gcs().send_text(MAV_SEVERITY_WARNING, "Failsafe trigger 0x%x", static_cast<uint32_t>(failsafe.triggered));
+        gcs().send_text(MAV_SEVERITY_WARNING, "Failsafe trigger 0x%x", (unsigned int)failsafe.triggered);
```

Comment:
Couldn't this just be `static_cast<unsigned>`, on the basis that someone deemed the `static_cast` to be a good thing?

@khancyr is yet to show a case where `static_cast` is a huge win for us, however :-)

Similarly for the other places in this PR.

Reply:
I'm not a fan of the static_cast thing so because the cast is current incorrect I took the liberty of changing it to the form that I prefer.